src/aio.rs

```diff
@@ -577,3 +577,91 @@ impl ConnectionLike for MultiplexedConnection {
         self.db
     }
 }
+
+#[cfg(feature = "tokio-rt-core")]
+mod connection_manager {
+    use super::*;
+
+    /// A `ConnectionManager` is a proxy that wraps a multiplexed connection and
+    /// automatically reconnects to the server when necessary.
+    pub struct ConnectionManager {
+        connection_info: ConnectionInfo,
+        connection: MultiplexedConnection,
+    }
```

Comment:
The `ConnectionManager` wraps a `MultiplexedConnection`. I still have to add cloning behavior to match the `MultiplexedConnection`, but the idea was that the manager can be passed around to different tasks / connection handlers just like the `MultiplexedConnection` itself.